kyuubi-common/src/main/scala/org/apache/kyuubi/service/authentication/TokenAuthenticationProvider.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.service.authentication
+
+import javax.security.sasl.AuthenticationException
+
+trait TokenAuthenticationProvider {
+
+  /**
+   * The authenticate method is called by the Kyuubi Server authentication layer
+   * to authenticate users for their requests.
+   * If the token is to be granted, return nothing/throw nothing.
+   * When the token is to be disallowed, throw an appropriate [[AuthenticationException]].
+   *
+   * @param token The token received over the connection request
+   *
+   * @throws AuthenticationException When the token is found to be invalid by the implementation
+   */
+  @throws[AuthenticationException]
+  def authenticate(token: String): Unit
```

Comment:
def authenticate(credential: TokenCredential): Principal

Reply:
Please follow the same pattern likes https://github.com/apache/celeborn/blob/main/spi/src/main/java/org/apache/celeborn/spi/authentication/TokenAuthenticationProvider.java.